doc/source/programs/gdal_raster_stack.rst

```diff
@@ -0,0 +1,143 @@
+      -h, --help                                           Display help message and exit
+      --version                                            Display GDAL version and exit
+      --json-usage                                         Display usage as JSON document and exit
+      --drivers                                            Display driver list as JSON document and exit
+      --config <KEY>=<VALUE>                               Configuration option [may be repeated]
+      --progress                                           Display progress bar
+
+    Options:
+      -f, --of, --format, --output-format <OUTPUT-FORMAT>  Output format
+      --co, --creation-option <KEY>=<VALUE>                Creation option [may be repeated]
+      -b, --band <BAND>                                    Specify input band(s) number. [may be repeated]
+      --overwrite                                          Whether overwriting existing output is allowed
+      --resolution <xres>,<yres>|average|highest|lowest>   Target resolution (in destination CRS units)
+      --bbox <BBOX>                                        Target bounding box as xmin,ymin,xmax,ymax (in destination CRS units)
+      --target-aligned-pixels                              Round target extent to target resolution
+      --srcnodata <SRCNODATA>                              Set nodata values for input bands. [1.. values]
+      --dstnodata <VRTNODATA>                              Set nodata values at the destination band level. [1.. values]
+      --hidenodata                                         Makes the destination band not report the NoData.
+
+
+
+Description
+-----------
+
+This program combine together input bands from GDAL datasets into a dataset, that can be
+either a virtual stack in the :ref:`VRT (Virtual Dataset) <raster.vrt>` format,
+or in a more conventional raster format such as GeoTIFF.
+
+All bands of each input file are added as separate
+output bands, unless :option:`-b` is specified to select a subset of them.
+
+The list of input GDAL datasets can be specified at the end
+of the command line or put in a text file (one filename per line) for very long lists.
+Wildcards '*', '?' or '['] of :cpp:func:`VSIGlob` can be used, even on files located
+on network file systems such as /vsis3/, /vsigs/, /vsiaz/, etc.
+
+The following options are available:
+
+.. include:: gdal_options/of_raster_create_copy.rst
+
+.. include:: gdal_options/co.rst
+
+.. include:: gdal_options/overwrite.rst
+
+.. option:: -b <band>
+
+    Select an input <band> to be processed. Bands are numbered from 1.
+    If input bands not set all bands will be added to the output.
+    Multiple :option:`-b` switches may be used to select a set of input bands.
+
+.. option:: --resolution {<xres,yres>|highest|lowest|average}
+
+    In case the resolution of all input files is not the same, the :option:`--resolution` flag
+    enables the user to control the way the output resolution is computed.
```

Comment:
I would suggest a default behavior of issuing an error if the resolutions are not the same and the user has not specified `--resolution`.

Reply:
only for stack, or for mosaic too ?